core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -290,24 +291,29 @@ class AdminManager(val config: KafkaConfig,
           throw new InvalidPartitionsException(s"Topic already has $oldNumPartitions partitions.")
         }
 
-        val newPartitionsAssignment = Option(newPartition.newAssignments).map(_.asScala.map(_.asScala.map(_.toInt))).map { assignments =>
-          val unknownBrokers = assignments.flatten.toSet -- allBrokerIds
-          if (unknownBrokers.nonEmpty)
-            throw new InvalidReplicaAssignmentException(
-              s"Unknown broker(s) in replica assignment: ${unknownBrokers.mkString(", ")}.")
-
-          if (assignments.size != numPartitionsIncrement)
-            throw new InvalidReplicaAssignmentException(
-              s"Increasing the number of partitions by $numPartitionsIncrement " +
-                s"but ${assignments.size} assignments provided.")
-
-          assignments.zipWithIndex.map { case (replicas, index) =>
-            existingAssignment.size + index -> replicas
+        val newPartitionsAssignment = Option(newPartition.assignments())
+          .map { createPartitionAssignments => createPartitionAssignments.asScala.map {
```

Comment:
I think we could make this a little easier to follow just by splitting the `None` case. For example:

```scala 
Option(newPartition.assignments) match {
 case None => CreatePartitionsMetadata(topic, None, ApiError.NONE)
 case Some(assignmentMap) =>
   ...
  CreatePartitionsMetadata(topic, updatedReplicaAssignment, ApiError.NONE)
}
```

Reply:
Merged the two map and refactored few lines. I think its better and more readable now.